Automattic/jetpack
# Components: Update Toolbar and ClipboardInput components to prevent deprecation notices

Fixes https://github.com/Automattic/vulcan/issues/562

## Proposed changes:

* This PR resolves several deprecation notices and warnings that were visible in the browser console:
  * There were several instances relating to the use of the Toolbar component. This involved changes to the Donations, Subscriptions, and Tiled Gallery blocks. It also involved changes to a shared EditButton component (used by Tiled Gallery, Simple Payments, and Story blocks).
  * The `ClipboardInput` component was modified to make use of the Jetpack `CopyToClipboard` component, as the previously used `ClipboardButton` component was deprecated. This did involve a slight style change, but brings the copy button in alignment with other features using that shared component.

Toolbar related notices:
```
Using Toolbar without label prop is deprecated since version 5.6. Please use ToolbarGroup component instead. See: https://developer.wordpress.org/block-editor/components/toolbar/

Using custom components as toolbar controls is deprecated since version 5.6. Please use ToolbarItem, ToolbarButton or ToolbarDropdownMenu components instead. See: https://developer.wordpress.org/block-editor/components/toolbar-button/#inside-blockcontrols
```
 ClipboardInput related notice:
```
wp.components.ClipboardButton is deprecated since version 5.8. Please use wp.compose.useCopyToClipboard instead.
```

### Other information:


## Jetpack product discussion

https://github.com/Automattic/vulcan/issues/562

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

You may need script debug enabled: `define( 'SCRIPT_DEBUG', true );` in `wp-config.php`.

In trunk:
* Add each of a Donations block, Subscriptions (Subscribe) block, and Tiled Gallery block.
* Click on each block, and you should see (for each) a notice in the console alluding to `Using custom components as toolbar controls is deprecated since version 5.6.`.
* Then add a Simple Payments and Story block, and as well as the Tiled Gallery block you'll see a notice alluding to `Using Toolbar without label prop is deprecated since version 5.6.`
* Lastly, click on the Jetpack icon in the top right of the editor to open the Jetpack sidebar. Expand the 'Shortlink' section, and you'll see a notice in the console alluding to `wp.components.ClipboardButton is deprecated since version 5.8`.

With this PR applied:
* Follow the same steps as above. The notices should not appear. The toolbar should appear (and function) just as it did before in each case. This refers specifically to the 'edit' option for Story, Simple Payments, and Tiled Gallery blocks. With Tiled Gallery it also refers to the 'Pick an image filter' button, with Donations it refers to the 'Change currency' button, and with Subscriptions (Subscribe) it refers to the 'Set up a paid plan' button.
* When expanding the Shortlink section, the style and behavior of the copy button is different as it uses a new component. Make sure this works as intended.
* Note - you may see other notices related to aria-hidden when clicking on blocks - issue created here: https://github.com/Automattic/vulcan/issues/572

**WoA / Simple**

It would be good to test in these environments as well to make sure the touched blocks (and Shortlinks) continue to work as expected.
